Fix env variables in `convex.yaml`

hahagu opened the pull request:
## Changes
- Fix env variables `CONVEX_CLOUD_ORIGIN` and `CONVEX_SITE_ORIGIN` to be proper values.
Otherwise, you will get a 404 error running queries/mutations from actions.

- Also added nginx with `SERVICE_URL_SITE` for site url support. Without this, http actions will have to go to `${SERVICE_URL_BACKEND}/http` 

## Issues
- fix #5569
- fix #7232

hahagu committed:
Fix env variables in `convex.yaml`

Fix env variables `CONVEX_CLOUD_ORIGIN` and `CONVEX_SITE_ORIGIN` to be proper values.

Otherwise, you will get a 404 error running queries/mutations from actions.

Also added `SERVICE_URL_SITE_3211` for site url support. Without this, http actions will have to go to `${SERVICE_URL_BACKEND}/http`

hahagu committed:
Merge branch 'next' into patch-1

orenaksakal reviewed (approved):
![sacha-baron-cohen-thumbs-up-gif-by-amazon-prime-video](https://github.com/user-attachments/assets/20c3e643-eaa7-4757-bcae-958143350830)

hahagu committed:
Update environment variables in convex.yaml

Commented out SERVICE_URL_SITE_3211 and updated CONVEX_SITE_ORIGIN note.

hahagu committed:
Merge branch 'next' into patch-1

hahagu commented:
There's currently no way to have a single service with two ports have urls generated for on coolify.
Therefore, I have commented out `SERVICE_URL_SITE_3211`.

hahagu committed:
Update `convex.yaml` template for site configuration

There's currently no way to have a single service with two ports have urls generated for on coolify.

So, we make use of an nginx proxy for the site.

hahagu committed:
Update environment variables in convex.yaml

Fix accidentally edited variables in convex.yaml

Cinzya commented:
> There's currently no way to have a single service with two ports have urls generated for on coolify.
Therefore, I have commented out SERVICE_URL_SITE_3211.

@hahagu does this mean users need to add those two URLs with their dedicated ports manually? Or did you find a different workaround? Would be good if you could include that information to the Coolify Docs, so people understand better what to do.

Cinzya commented on templates/compose/convex.yaml:
```diff
@@ -7,19 +7,17 @@
 services:
   backend:
-    image: ghcr.io/get-convex/convex-backend:00bd92723422f3bff968230c94ccdeb8c1719832
+    image: ghcr.io/get-convex/convex-backend:latest
```
Please pin it to a specific version or hash.

Cinzya commented on templates/compose/convex.yaml:
```diff
@@ -46,16 +44,42 @@ services:
+        condition: service_healthy
+
+volumes:
+  convex-data:
```
this is not needed. Coolify will add that already when parsing the compose file.

hahagu committed:
Merge branch 'next' into patch-1

hahagu committed:
Fix version to hash and remove volume config

hahagu commented:
> > There's currently no way to have a single service with two ports have urls generated for on coolify.
> > Therefore, I have commented out SERVICE_URL_SITE_3211.
> 
> @hahagu does this mean users need to add those two URLs with their dedicated ports manually? Or did you find a different workaround? Would be good if you could include that information to the Coolify Docs, so people understand better what to do.

Implemented a nginx proxy as a workaround.

hahagu committed:
Merge branch 'next' into patch-1

hahagu commented:
@Cinzya Please review, thanks!

KazBrekker1 commented:
I tried the yml but i faced an issue opening the dashboard, for some reason initially it shows that the backend is not running, and then after the following log errors, it connects successfully.
<img width="916" height="99" alt="image" src="https://github.com/user-attachments/assets/4ae4b974-587b-4d96-9508-b6782049c707" />

peaklabs-dev commented:
The fact that we cannot generate multiple domains with different ports for one container is most likely a parser bug. I have let Andras know internally and once the bug is fixed the template can be properly updated without an additional proxy.
